pommerman/envs/v3.py

```diff
@@ -0,0 +1,110 @@
+"""The Pommerman v3 Environment, which implements Sudden Death mode.
+
+The game ends when any agent dies. In FFA, all other agents win. In any of the team
+modes, the team with the most remaining players wins. If all players die at the same
+time or the same number of players die on each time, the game ends in a tie. All
+players/teams win if the game ends without anyone dying.
+"""
+from .. import constants
+from .. import forward_model
+from .. import utility
+from . import v0
+
+
+class Pomme(v0.Pomme):
+    metadata = {
+        'render.modes': ['human', 'rgb_array']
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _get_done(self):
+        alive = [agent for agent in self._agents if agent.is_alive]
+        alive_ids = sorted([agent.agent_id for agent in alive])
+        if self._step_count >= self._max_steps:
+            return True
+        elif len(alive) < 4:
+            return True
+        return False
+
+    def _get_info(self, done, rewards):
+        if self._game_type == constants.GameType.FFA:
+            alive = [agent for agent in self._agents if agent.is_alive]
+            if done and len(alive) == 0:
+                return {
+                    'result': constants.Result.Tie,
+                }
+            elif done:
+                return {
+                    'result': constants.Result.Win,
+                    'winners': [num for num, reward in enumerate(rewards) if reward == 1]
+                }
+            else:
+                return {'result': constants.Result.Incomplete}
+        elif self._game_type == constants.GameType.Team or \
+                                self._game_type == constants.GameType.TeamRadio:
+            alive = [agent for agent in self._agents if agent.is_alive]
+            if done and len(alive) < 4:
+                team1 = []
+                team2 = []
+                for agent in alive:
+                    if agent.agent_id == 0 or agent.agent_id == 2:
+                        team1.append(agent.agent_id)
+                    else:
+                        team2.append(agent.agent_id)
+                if len(team1) == len(team2):
+                    return {
+                        'result': constants.Result.Tie,
+                    }
+                else:
+                    return {
+                        'result': constants.Result.Win,
+                        'winners': [num for num, reward in enumerate(rewards) if reward == 1]
+                    }
+            else:
+                return {'result': constants.Result.Incomplete}
+        elif done:
```

Comment:
Will this ever be reached?

Reply:
I copied the format from v0, which also has some else statements that don't appear reachable. I can remove it if you wish.

Reply:
Removing.